pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -625,6 +627,12 @@ func ConsolidateObjects(env api.Environment, cr *api.KieApp) api.Environment {
 	for index := range env.Servers {
 		serverSet, _ := GetServerSet(cr, index)
 		env.Servers[index] = ConstructObject(env.Servers[index], serverSet.KieAppObject)
+		// apply the build config envs provided through cr to the given kieSerever BuildConfig Spec envs
+		for bcindex := range env.Servers[index].BuildConfigs {
+			env.Servers[index].BuildConfigs[bcindex].Spec.Strategy.SourceStrategy.Env = shared.EnvOverride(
+				env.Servers[index].BuildConfigs[bcindex].Spec.Strategy.SourceStrategy.Env,
+				cr.Spec.Objects.Servers[index].Build.Env)
```

Comment:
i believe this can be `serverSet.Build.Env`

Reply:
let me try.